azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/dag/ErroredDependencyTaskException.java

```diff
@@ -0,0 +1,14 @@
+/**
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ *
+ */
+
+package com.microsoft.azure.management.resources.fluentcore.dag;
+
+/**
+ * An FaultedDependencyException is emitted when a task cannot be executed due to a faulted decedent task.
+ */
+final class ErroredDependencyTaskException extends Exception {
```

Comment:
~~Make it an unchecked exception~~
Actually, since this is only used internally and should not stop the program from running immediately, it should be a checked exception.